route/build/src/irf_route.f90

```diff
@@ -105,7 +110,14 @@ subroutine irf_route(iEns,          &  ! input: index of runoff ensemble to be p
      seg:do iSeg=1,river_basin(ix)%branch(iTrib)%nRch
        jSeg = river_basin(ix)%branch(iTrib)%segIndex(iSeg)
        if (.not. doRoute(jSeg)) cycle
-       call segment_irf(iEns, jSeg, ixDesire, NETOPO_IN, RCHFLX_out, ierr, cmessage)
+        if (NETOPO_in(jseg)%islake) then
+         print*, 'islake = ', NETOPO_in(jseg)%islake ! to check if it is lake
+         print*, 'reach id that is lake = ', NETOPO_in(jseg)%REACHID ! to check the reach id of lake
+         call lake_route(iEns, jSeg, ixDesire, NETOPO_in, RPARAM_in, RCHFLX_out, ierr, message)
```

Comment:
Shervan,  you will need to add RPARAM_in in OMP specification - `!$OMP   shared(RPARAM_in)`.  See before tributary loop.

Reply:
added, I still need to check if it compiles. also we add the job with more node (currently only one node).

Reply:
yes, it would be good to run with several nodes at this point.  if you do that, make sure that changing number of node (also called MPI tasks) won't change the results.  nccmp command can be used to compare two netcdfs.   Note that changing number of node will change reach and hru orders in output (this is related to PIO).  I can send you small python script to reorder one netcdf to match with order of another.

Reply:
would be awesome to have your python code to order the segments. I have nccmp for comparison.

Reply:
the current code (which changed today to read RATECVA and B) results in the same lake volume as from from the changes before. I can do a more detailed checked for each reach as well.
![lake_vol_new_3node](https://user-images.githubusercontent.com/28273583/78089337-5fe5e280-7384-11ea-9ae0-ffaf27dee496.jpg)

Reply:
oh, because you have not used rating curve parameters in lake yet?

Reply:
yes, exactly. it just make sure that the code runs and give the similar result. the parameters are passed to the subroutine. The next steps are:
1- using the parameters for stage discharge relationship (including making sure storage is not negative, etc etc)
2- add the in-basin routing output to the lake (maybe we should skip the in-basin routing when lake flag is on)